ehrapy/api/io/read.py

```diff
@@ -317,56 +382,112 @@ def _decode_cached_adata(adata: AnnData, column_obs_only: List[str]) -> AnnData:
+
+                .. code-block:: python
+                       # some setup code here
+                       ...
+                       # filename
+                       identifier1 = "MyFile"
+                       identifier2 = "MyOtherFile"
+                       # identifier1 is in the index and columns_obs_only keys
+                       # -> index column will be MyColumn1 and columns_obs_only will include MyColumn2 and MyColumn3
+                       index_columns = {"MyFile":["MyColumn1"]}
+                       columns_obs_only = {"MyFile":["MyColumn2", "MyColumn3"]}
 
         Args:
-            df: Pandas DataFrame to set the index for
-            index_column: The name of the index column
+            identifier: The name of the
+            index_columns: Index columns
+            columns_obs_only: Columns for obs only
 
         Returns:
-            An :class:`~pd.DataFrame` object
+            Index column (if any) and columns obs only (if any) for this specific AnnData object
         """
-        column_names = list(df.columns)
-        if isinstance(index_column, str):
-            df = df.set_index(index_column)
-        elif isinstance(index_column, int):
-            df = df.set_index(column_names[index_column])
-        else:
-            if "patient_id" == column_names[0]:
-                df = df.set_index("patient_id")
-            else:
-                if not DataReader.suppress_warnings:
-                    warnings.warn(
-                        "Did not find patient_id column at column 0 and no index column was passed. "
-                        "Using default, numerical indices instead!",
-                        IndexColumnWarning,
-                    )
+        _index_column = None
+        _columns_obs_only = None
+        # get index column (if any)
+        if index_columns and identifier in index_columns.keys():
+            _index_column = index_columns[identifier]
+        elif index_columns and "default" in index_columns.keys():
+            _index_column = index_columns["default"]
+
+        # get columns obs only (if any)
+        if columns_obs_only and identifier in columns_obs_only.keys():
+            _columns_obs_only = columns_obs_only[identifier]
+        elif columns_obs_only and "default" in columns_obs_only.keys():
+            _columns_obs_only = columns_obs_only["default"]
+
+        # if there is only one obs only column, it might have been passed as single string
+        if isinstance(_columns_obs_only, str):
+            _columns_obs_only = [_columns_obs_only]
+
+        # if index column is also found in column_obs_only, use default indices instead and only move it to obs only, but warn the user
+        if _index_column and _index_column in _columns_obs_only:
+            print(
+                f"[bold yellow]Index column [bold blue]{_index_column} [bold yellow]for file [bold blue]{identifier} [bold yellow]is also used as a column "
```

Comment:
Pretty sure that you can actually remove the bolds after the first bold. Changing the color should suffice.

Reply:
Will check this out